en/docs/chapter_hashing/hash_map.md

```diff
@@ -1,30 +1,30 @@
 # Hash table
 
-A <u>hash table</u> achieves efficient element querying by establishing a mapping between keys and values. Specifically, when we input a `key` into the hash table, we can retrieve the corresponding `value` in $O(1)$ time.
+A <u>hash table</u>, also known as a <u>hash map</u>, is a data structure that establishes a mapping between keys and values, enabling efficient element retrieval. Specifically, when we input a `key` into the hash table, we can retrive the corresponding `value` in $O(1)$ time complexity.
 
-As shown in the figure below, given $n$ students, each with two pieces of data: "name" and "student number". If we want to implement a query feature that returns the corresponding name when given a student number, we can use the hash table shown in the figure below.
+As shown in the diagram below, given $n$ students, each student has two data fields: "Name" and "Student ID". If we want to implement a query function that takes a student ID as input and returns the corresponding name, we can use the hash table depicted below.
 
 ![Abstract representation of a hash table](hash_map.assets/hash_table_lookup.png)
 
-Apart from hash tables, arrays and linked lists can also be used to implement querying functions. Their efficiency is compared in the table below.
+In addition to hash tables, arrays and linked lists can also be used to implement query functionality, but the time complexity is different. The table below shows the comparison of time efficiency for common operations among them:
 
-- **Adding elements**: Simply add the element to the end of the array (or linked list), using $O(1)$ time.
-- **Querying elements**: Since the array (or linked list) is unordered, it requires traversing all the elements, using $O(n)$ time.
-- **Deleting elements**: First, locate the element, then delete it from the array (or linked list), using $O(n)$ time.
+- **Inserting an element**: Simply append the element to the tail of the array (or linked list). The time complexity of this operation is $O(1)$.
+- **Searching for an element**: As the array (or linked list) is unsorted, searching for an element requires traversing through all of the elements. The time complexity of this operation is $O(n)$.
+- **Deleting an element**: In order to remove elements, we need to locate them first. Then, delete them from the array (or linked list). The time complexity of this operation is $O(n)$.
```

Comment:
Starting a sentence with "Then, ..." is controversial (It's grammatically OK, but feels sudden/jarring). Just a style suggestion but maybe something like "To remove elements, we first need to locate them. After locating the elements, we delete them from the array (or linked list)..."

Reply:
Thank you for your suggestion. However, I think changing from "Then...." to duplicated words "locate..." might not be a good idea. How about something like "To remove elements, we first need to locate them. Subsequently, we delete..."?

Reply:
Yes it sounds better. I've changed the sentence to singular form. Have a check. https://github.com/krahets/hello-algo/pull/1358/commits/8b39df276d3d6c05ddec9c38013ce7baf5f8420b